caddyhttp/fastcgi/fastcgi.go

```diff
@@ -86,16 +86,12 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 			var fcgiBackend *FCGIClient
 			reuse := false
 			if usePersistentFcgiConnections {
-				metaMutex.RLock()
+				metaMutex.Lock()
 				// re use connection, if possible
 				if _, ok := poorMansSerialisation[network+address]; !ok {
-					metaMutex.RUnlock()
-					metaMutex.Lock()
 					poorMansSerialisation[network+address] = new(sync.Mutex)
-					metaMutex.Unlock()
-					metaMutex.RLock()
```

Comment:
This whole lock/unlock was not only ugly, unlocking in between or allowing for multiple reads at that point was not a good idea, anyway. ;)